src/include/platform/internal/GenericPlatformManagerImpl_CMSISOS.ipp

```diff
@@ -0,0 +1,400 @@
+ */
+
+#ifndef GENERIC_PLATFORM_MANAGER_IMPL_CMSISOS_CPP
+#define GENERIC_PLATFORM_MANAGER_IMPL_CMSISOS_CPP
+
+#include <platform/PlatformManager.h>
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+#include <platform/internal/GenericPlatformManagerImpl_CMSISOS.h>
+
+#include <lib/support/CodeUtils.h>
+
+// Include the non-inline definitions for the GenericPlatformManagerImpl<> template,
+// from which the GenericPlatformManagerImpl_CMSISOS<> template inherits.
+#include <platform/internal/GenericPlatformManagerImpl.ipp>
+
+namespace chip {
+namespace DeviceLayer {
+namespace Internal {
+
+template <class ImplClass>
+CHIP_ERROR GenericPlatformManagerImpl_CMSISOS<ImplClass>::_InitChipStack(void)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    mNextTimerBaseTime      = osKernelGetTickCount();
+    mNextTimerDurationTicks = 0;
+    mChipTimerActive        = false;
+
+    // We support calling Shutdown followed by InitChipStack, because some tests
+    // do that.  To keep things simple for existing consumers, we keep not
+    // destroying our lock and queue in shutdown, but rather check whether they
+    // already exist here before trying to create them.
+    if (mChipStackLock == nullptr)
+    {
+        mChipStackLock = osMutexNew(&mChipStackMutexAttr);
+        if (mChipStackLock == nullptr)
+        {
+            ChipLogError(DeviceLayer, "Failed to create CHIP stack lock");
+            ExitNow(err = CHIP_ERROR_NO_MEMORY);
+        }
+    }
+
+    if (mChipEventQueue == nullptr)
+    {
+        mChipEventQueue = osMessageQueueNew(CHIP_DEVICE_CONFIG_MAX_EVENT_QUEUE_SIZE, sizeof(ChipDeviceEvent), mEventQueueAttrPtr);
+        if (mChipEventQueue == nullptr)
+        {
+            ChipLogError(DeviceLayer, "Failed to allocate CHIP main event queue");
+            ExitNow(err = CHIP_ERROR_NO_MEMORY);
+        }
+    }
+    else
+    {
+        // Clear out any events that might be stuck in the queue, so we start
+        // with a clean slate, as if we had just re-created the queue.
+        osMessageQueueReset(mChipEventQueue);
+    }
+
+    mShouldRunEventLoop.store(false);
+
+#if defined(CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING) && CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING
+    if (mBackgroundEventQueue == nullptr)
+    {
+        mBackgroundEventQueue =
+            osMessageQueueNew(CHIP_DEVICE_CONFIG_BG_MAX_EVENT_QUEUE_SIZE, sizeof(ChipDeviceEvent), mBgQueueAttrPtr);
+        if (mBackgroundEventQueue == nullptr)
+        {
+            ChipLogError(DeviceLayer, "Failed to allocate CHIP background event queue");
+            ExitNow(err = CHIP_ERROR_NO_MEMORY);
+        }
+    }
+    else
+    {
+        osMessageQueueReset(mBackgroundEventQueue);
+    }
+
+    mShouldRunBackgroundEventLoop.store(false);
+#endif
+
+    // Call up to the base class _InitChipStack() to perform the bulk of the initialization.
+    err = GenericPlatformManagerImpl<ImplClass>::_InitChipStack();
+    SuccessOrExit(err);
+
+exit:
```

Comment:
To avoid having jumps, we could just use VerifyOrReturnError which would simplify the return statemens since there is not clean up in the exit block.